main.go

```diff
@@ -2,14 +2,59 @@ package main
 
 import (
 	"context"
+	"encoding/json"
+	"fmt"
+	"io"
 	"os"
 
 	"github.com/databricks/cli/cmd"
 	"github.com/databricks/cli/cmd/root"
+	"github.com/databricks/cli/libs/telemetry"
+	"github.com/databricks/databricks-sdk-go/logger"
 )
 
 func main() {
 	ctx := context.Background()
+
+	// A new process is spawned for uploading telemetry data. We handle this separately
+	// from the rest of the CLI commands.
+	// This is done because [root.Execute] spawns a new process to run the
+	// "telemetry upload" command if there are logs to be uploaded. Having this outside
+	// of [root.Execute] ensures that the telemetry upload process is not spawned
+	// infinitely in a recursive manner.
```

Comment:
From top of my head this makes it incompatible with inprocess testing of acceptance tests. There could be other consequences of having this implemented in a special way. For example, to debug telemetry I might want to pass --debug, this won't work out of the box either.

On the other hand preventing infinite recursion seems like a straightforward thing, just check that current command is not 'telemetry'.

Reply:
That's true, but I'd much rather avoid this class of issues altogether. It's difficult to write a robust end-to-end test that ensures infinite recursion is not happening, and this approach guarantees that it's not even a possibility. The outage, if we end up with infinite recursion, is pretty massive since we'd essentially brick machines that use the CLI due to excessive resource consumption.

Other reasons that I prefer this approach:
1. It's easier to reason about the control flow across processes. Using Cobra, you get a lot of intermediate code, which we don't have to reason about here.
2. We don't really need customer telemetry debug logs. They'll never report that telemetry is not working for them.

Reply:
> Using Cobra, you get a lot of intermediate code,

Could you elaborate, what intermediate code?

> We don't really need customer telemetry debug logs. They'll never report that telemetry is not working for them. These logs will also never be visible to the customer.

But we ourselves might want to debug it. Or tell field engineers how to debug it.

Reply:
Discussed offline. We'll go ahead with the current approach for now, and change this to a cobra command if needed in the future.